appveyor.yml

```diff
@@ -28,23 +28,26 @@ before_build:
         }
     - ps: $env:VSCOMNTOOLS=(Get-Content ("env:VS" + "$env:VSVER" + "0COMNTOOLS"))
     - call "%VSCOMNTOOLS%\..\..\VC\vcvarsall.bat" %VCVARS_PLATFORM%
-    - mkdir _build
-    - cd _build
-    - perl ..\Configure %TARGET% %SHARED%
-    - cd ..
     - ps: >-
         if (-not $env:APPVEYOR_PULL_REQUEST_NUMBER`
             -or (&git log -2 | Select-String "\[extended tests\]") ) {
             $env:EXTENDED_TESTS="yes"
+            if ($env:Configuration -NotMatch "shared") {
+                exit 0
+            }
         }
+    - mkdir _build
+    - cd _build
+    - perl ..\Configure %TARGET% %SHARED%
+    - cd ..
 
 build_script:
+    - cd _build
     - ps: >-
         If ($env:Configuration -Match "shared" -or $env:EXTENDED_TESTS) {
-            cd _build
             cmd /c "nmake 2>&1"
-            cd ..
         }
+    - cd ..
```

Comment:
:-) As already said, "it might be worth keeping in mind that powershell is almost, but not quite, entirely unlike the kind of shell you'd imagine when you hear "shell" in its name."